server/storage/backend/tx_buffer.go

```diff
@@ -42,6 +42,9 @@ type txWriteBuffer struct {
 	// Map from bucket ID into information whether this bucket is edited
 	// sequentially (i.e. keys are growing monotonically).
 	bucket2seq map[BucketID]bool
+
+	// store revision for keys bucket
+	bufRev int64
```

Comment:
I'm not supper familiar with MVCC implementation, but adding revision  here doesn't seem right. This change introduces concept of revision from mvcc into generic backend, that should not have any assumption about keys. Sometimes we need to break nice abstractions to gain performance, but I would prefer if we ensure that it cannot be avoided.

Reply:
```
func (baseReadTx *baseReadTx) UnsafeRangeWithLock(bucketType Bucket, key, endKey []byte, limit int64, b Backend) ([][]byte, [][]byte) {
    ...
    baseReadTx.mu.RLock()
    if keyRev >= baseReadTx.buf.bufMinRev || endKeyRev >= baseReadTx.buf.bufMinRev {
        keys, vals = baseReadTx.buf.Range(bucketType, key, endKey, limit)
        ...
    }
    baseReadTx.mu.RUnlock()
    ...
}

Revision can simply avoid ranging in the baseReadTx.buf, which decreaseing lock time.

And in the etcd backend, revision is the key for bolt db in key bucket, so generic backend just has revision already.
```